compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/selection/SelectionGestures.desktop.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text.selection
+
+import androidx.compose.foundation.DesktopPlatform
+import androidx.compose.foundation.contextMenuOpenDetector
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.pointer.PointerEventPass
+import androidx.compose.ui.layout.LayoutCoordinates
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextRange
+
+
+private inline fun Modifier.onMacOsOnly(crossinline modifier: Modifier.() -> Modifier) = this.then(
+    if (DesktopPlatform.Current == DesktopPlatform.MacOS) {
+        this.modifier()
```

Comment:
`this.then(this.modifier())` - won't it duplicate `this`?

Reply:
🤦